File: Poly.app/poly.dev/api/src/main/java/ru/polydev/poly_dev/dto/request/SignUpRequest.java
package ru.polydev.poly_dev.dto.request;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@AllArgsConstructor
@Getter
@Setter
public class SignUpRequest {
    private String firstName;
    private String lastName;
    private String middleName;
    private String institute;
    private String direction;
    private String course;
    private String email;
    private String password;
    private String vk;
    private String tg;
}

File: Poly.app/poly.dev/app/lib/repositories/dto/sign_up_request.dart
class SignUpRequest {
  SignUpRequest(
    this.firstName,
    this.lastName,
    this.middleName,
    this.institute,
    this.direction,
    this.course,
    this.email,
    this.password,
    this.vk,
    this.tg,
  );

  final String firstName;
  final String lastName;
  final String middleName;
  final String institute;
  final String direction;
  final String course;
  final String email;
  final String password;
  String? vk;
  String? tg;

  Map<String, dynamic> toJson() {
    return {
      "firstName": firstName,
      "lastName": lastName,
      "middleName": middleName,
      "institute": institute,
      "direction": direction,
      "course": course,
      "email": email,
      "password": password,
      "vk": vk,
      "tg": tg,
    };
  }
}
